sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisCheckpointStore.java

```diff
@@ -211,18 +266,30 @@ public Flux<PartitionOwnership> listOwnership(String fullyQualifiedNamespace, St
      * This method updates the checkpoint in the Jedis resource for a given partition.
      *
      * @param checkpoint Checkpoint information for this partition
+     *
      * @return Mono that completes if no errors take place
+     * @throws NullPointerException if {@code checkpoint} is null.
+     * @throws IllegalArgumentException if {@code checkpoint} does not have a sequenceNumber or offset.
      */
     @Override
     public Mono<Void> updateCheckpoint(Checkpoint checkpoint) {
+        if (Objects.isNull(checkpoint)) {
+            return Mono.error(new NullPointerException("'checkpoint' cannot be null."));
+        }
+
         if (!isCheckpointValid(checkpoint)) {
-            throw LOGGER.logExceptionAsWarning(Exceptions
-                .propagate(new IllegalStateException(
+            return Mono.error(addEventHubInformation(LOGGER.atError(), checkpoint.getFullyQualifiedNamespace(),
+                checkpoint.getEventHubName(), checkpoint.getConsumerGroup())
+                .addKeyValue(PARTITION_ID_KEY, checkpoint.getPartitionId())
+                .log(new IllegalArgumentException(
                     "Checkpoint is either null, or both the offset and the sequence number are null.")));
```

Comment:
nit, `FluxUtils.monoError` might be a bit nicer:
```suggestion
            return FluxUtils.monoError(addEventHubInformation(LOGGER.atError(), checkpoint.getFullyQualifiedNamespace(),
                    checkpoint.getEventHubName(), checkpoint.getConsumerGroup())
                    .addKeyValue(PARTITION_ID_KEY, checkpoint.getPartitionId()), 
                new IllegalArgumentException("Checkpoint is either null, or both the offset and the sequence number are null."));
```